### ISSUE TITLE
🐛 [firebase_app_check]  ReCAPTCHA error. (appCheck/recaptcha-error). on Flutter web

### ISSUE THREAD

**Issue by rahmanrezaee** [firebase/flutterfire#12226]
## Bug report

**Describe the bug**
after upgrading the flutter SDK to (3.16.9) the app check does not work further and shows an unclear error
[2024-01-31T07:02:05.184Z]  @firebase/app-check:
FirebaseError: AppCheck: ReCAPTCHA error. (appCheck/recaptcha-error). 

### Steps to reproduce
Not really clear to me how to reproduce, since it did not work for me from one day to another

### Expected behavior
AppCheck should work when it is configured right on Flutter web.

### Demo Code

```dart

import 'dart:async';

import 'package:firebase_app_check/firebase_app_check.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';
import 'constant/color_contant.dart';
import 'firebase_options.dart';

void main() async {
    WidgetsFlutterBinding.ensureInitialized();
  

    await Firebase.initializeApp(
        options: DefaultFirebaseOptions.currentPlatform);


    try {
      await FirebaseAppCheck.instance.activate(
        webProvider:
            ReCaptchaEnterpriseProvider(AppConstants.enterpriseRecapKey),
      );
    } on FirebaseException catch (e) {
      print(e.message);
    }

    if (kDebugMode) {
      final token = await FirebaseAppCheck.instance.getToken();
      print("app check token: $token");
    }
    runApp(....);
 
}

```

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Flutter Doctor Log</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.16.9, on Microsoft Windows [Version 10.0.19045.3930], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 2023.1)
[√] VS Code (version 1.85.2)
[√] Connected device (4 available)
[√] Network resources

! Doctor found issues in 1 category.
```

<details><summary>Console after running app</summary>
```
Launching lib\main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...
This app is linked to the debug service: ws://127.0.0.1:62811/fTK8LqT8HYU=/ws
Debug service listening on ws://127.0.0.1:62811/fTK8LqT8HYU=/ws
Debug service listening on ws://127.0.0.1:62811/fTK8LqT8HYU=/ws
The webOnlyAssetManager getter is deprecated and will be removed in a future release. Please use `assetManager` from `dart:ui_web` instead.
The webOnlyAssetManager getter is deprecated and will be removed in a future release. Please use `assetManager` from `dart:ui_web` instead.
[2024-01-31T07:02:05.184Z]  @firebase/app-check:
FirebaseError: AppCheck: ReCAPTCHA error. (appCheck/recaptcha-error).
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Dependancy</summary>

```
Dart SDK 3.2.6
Flutter SDK 3.16.9
course_platform 1.0.0+2

dependencies:
- animated_text_kit 4.2.2 [flutter characters]
- audio_video_progress_bar 2.0.1 [flutter]
- audioplayers 5.2.1 [audioplayers_android audioplayers_darwin audioplayers_linux audioplayers_platform_interface audioplayers_web audioplayers_windows file flutter http meta path_provider synchronized uuid]
- back_button_interceptor 7.0.0 [collection flutter]
- beamer 1.6.0 [flutter flutter_web_plugins]
- carousel_slider 4.2.1 [flutter]
- chewie 1.7.4 [cupertino_icons flutter provider video_player wakelock_plus]
- cloud_firestore 4.14.0 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- confetti 0.7.0 [flutter vector_math]
- cupertino_icons 1.0.6
- dropdown_button2 2.3.9 [flutter meta]
- dropdown_search 5.0.6 [flutter]
- dynamic_height_grid_view 0.0.4 [flutter]
- email_validator 2.1.17
- firebase_app_check 0.2.1+8 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 4.16.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.24.2 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_storage 11.6.0 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math web sky_engine]
- flutter_highlight 0.7.0 [flutter highlight]
- flutter_inappwebview 6.0.0 [flutter flutter_inappwebview_platform_interface flutter_inappwebview_android flutter_inappwebview_ios flutter_inappwebview_macos flutter_inappwebview_web]
- flutter_layout_grid 2.0.5 [flutter collection meta quiver]
- flutter_linkify 6.0.0 [flutter linkify]
- flutter_riverpod 2.4.9 [collection flutter meta riverpod state_notifier]
- flutter_stripe 9.6.0 [flutter meta stripe_android stripe_ios stripe_platform_interface]
- flutter_stripe_web 4.6.0 [flutter flutter_web_plugins freezed_annotation stripe_platform_interface js stripe_js]
- flutter_svg 2.0.9 [flutter vector_graphics vector_graphics_codec vector_graphics_compiler]
- flutter_widget_from_html 0.14.11 [flutter flutter_widget_from_html_core fwfh_cached_network_image fwfh_chewie fwfh_just_audio fwfh_svg fwfh_url_launcher fwfh_webview html]
- fluttertoast 8.2.4 [flutter flutter_web_plugins]
- font_awesome_flutter 10.6.0 [flutter]
- fwfh_text_style 2.23.8 [flutter]
- google_fonts 6.1.0 [flutter http path_provider crypto]
- highlight_text 1.7.0 [flutter]
- http 1.2.0 [async http_parser meta web]
- http_auth 1.0.4 [http crypto convert]
- image_downloader_web 2.0.4 [flutter flutter_web_plugins http universal_html]
- intl 0.19.0 [clock meta path]
- js 0.6.7 [meta]
- meta_seo 3.0.8 [flutter]
- pdf 3.10.7 [archive barcode bidi crypto image meta path_parsing vector_math xml]
- pointer_interceptor 0.10.1 [flutter flutter_web_plugins pointer_interceptor_ios pointer_interceptor_platform_interface pointer_interceptor_web]
- printing 5.11.1 [ffi flutter flutter_web_plugins http image js meta pdf plugin_platform_interface]
- qr_flutter 4.1.0 [flutter qr]
- razorpay_web 1.4.2 [flutter flutter_web_plugins eventify package_info_plus universal_platform]
- scroll_to_index 3.0.1 [flutter]
- seo 0.0.6 [flutter rxdart]
- shared_preferences 2.2.2 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- stack_trace 1.11.1 [path]
- super_banners 0.0.1 [flutter]
- url_launcher 6.2.3 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- video_player 2.8.2 [flutter html video_player_android video_player_avfoundation video_player_platform_interface video_player_web]
- video_player_web 2.1.3 [flutter flutter_web_plugins video_player_platform_interface]
- visibility_detector 0.4.0+2 [flutter]
- webview_flutter 4.4.4 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]

dev dependencies:
- flutter_lints 3.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection material_color_utilities meta source_span stream_channel string_scanner term_glyph web] 

transitive dependencies:
- _flutterfire_internals 1.3.16 [collection firebase_core firebase_core_platform_interface flutter meta]
- archive 3.4.10 [crypto path pointycastle]
- args 2.4.2
- async 2.11.0 [collection meta]
- audio_session 0.1.18 [flutter flutter_web_plugins rxdart meta]
- audioplayers_android 4.0.3 [audioplayers_platform_interface flutter]
- audioplayers_darwin 5.0.2 [audioplayers_platform_interface flutter]
- audioplayers_linux 3.1.0 [audioplayers_platform_interface flutter]
- audioplayers_platform_interface 6.1.0 [flutter meta plugin_platform_interface]
- audioplayers_web 4.1.0 [audioplayers_platform_interface flutter flutter_web_plugins js]
- audioplayers_windows 3.1.0 [audioplayers_platform_interface flutter]
- barcode 2.2.6 [meta qr]
- bidi 2.0.10
- boolean_selector 2.1.1 [source_span string_scanner]
- cached_network_image 3.3.1 [cached_network_image_platform_interface cached_network_image_web flutter flutter_cache_manager octo_image]
- cached_network_image_platform_interface 4.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.1.1 [cached_network_image_platform_interface flutter flutter_cache_manager]
- characters 1.3.0
- charcode 1.3.1
- clock 1.1.1
- cloud_firestore_platform_interface 6.1.0 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.9.0 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.18.0
- convert 3.1.1 [typed_data]
- crypto 3.0.3 [typed_data]
- csslib 1.0.0 [source_span]
- dbus 0.7.10 [args ffi meta xml]
- eventify 1.0.1
- fake_async 1.3.1 [clock collection]
- ffi 2.1.0
- file 7.0.0 [meta path]
- firebase_app_check_platform_interface 0.1.0+10 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_app_check_web 0.1.0+10 [_flutterfire_internals firebase_app_check_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_platform_interface 7.0.9 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.8.13 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core_platform_interface 5.0.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.10.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_storage_platform_interface 5.1.3 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 3.6.17 [_flutterfire_internals async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- fixnum 1.1.0
- flutter_cache_manager 3.3.1 [clock collection file flutter http path path_provider rxdart sqflite uuid]
- flutter_inappwebview_android 1.0.12 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_internal_annotations 1.1.1
- flutter_inappwebview_ios 1.0.13 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_macos 1.0.11 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_platform_interface 1.0.10 [flutter flutter_inappwebview_internal_annotations plugin_platform_interface]
- flutter_inappwebview_web 1.0.8 [flutter flutter_web_plugins js flutter_inappwebview_platform_interface]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- flutter_widget_from_html_core 0.14.11 [csslib flutter html logging]
- freezed_annotation 2.4.1 [collection json_annotation meta]
- fwfh_cached_network_image 0.14.2 [cached_network_image flutter flutter_cache_manager flutter_widget_from_html_core]
- fwfh_chewie 0.14.7 [chewie flutter flutter_widget_from_html_core video_player]
- fwfh_just_audio 0.14.2 [flutter flutter_widget_from_html_core just_audio]
- fwfh_svg 0.8.1 [flutter flutter_svg flutter_widget_from_html_core]
- fwfh_url_launcher 0.9.0+4 [flutter flutter_widget_from_html_core url_launcher]
- fwfh_webview 0.14.8 [flutter flutter_widget_from_html_core logging webview_flutter webview_flutter_android webview_flutter_wkwebview]
- highlight 0.7.0 [collection]
- html 0.15.4 [csslib source_span]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 4.1.4 [archive meta xml]
- json_annotation 4.8.1 [meta]
- just_audio 0.9.36 [just_audio_platform_interface just_audio_web audio_session rxdart path path_provider async uuid crypto meta flutter]
- just_audio_platform_interface 4.2.2 [flutter plugin_platform_interface]
- just_audio_web 0.4.9 [just_audio_platform_interface flutter flutter_web_plugins]
- linkify 5.0.0
- lints 3.0.0
- logging 1.2.0
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.10.0
- nested 1.0.0 [flutter]
- octo_image 2.0.0 [flutter]
- package_info_plus 4.2.0 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface win32]
- package_info_plus_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- path 1.8.3
- path_parsing 1.0.1 [vector_math meta]
- path_provider 2.1.2 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- path_provider_android 2.2.2 [flutter path_provider_platform_interface]
- path_provider_foundation 2.3.2 [flutter path_provider_platform_interface]
- path_provider_linux 2.2.1 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.1.2 [flutter platform plugin_platform_interface]
- path_provider_windows 2.2.1 [ffi flutter path path_provider_platform_interface win32]
- petitparser 6.0.2 [meta]
- platform 3.1.4
- plugin_platform_interface 2.1.8 [meta]
- pointer_interceptor_ios 0.10.0+2 [flutter plugin_platform_interface pointer_interceptor_platform_interface]
- pointer_interceptor_platform_interface 0.10.0+1 [flutter plugin_platform_interface]
- pointer_interceptor_web 0.10.1+2 [flutter flutter_web_plugins plugin_platform_interface pointer_interceptor_platform_interface web]
- pointycastle 3.7.4 [collection convert js]
- provider 6.1.1 [collection flutter nested]
- qr 3.0.1 [meta]
- quiver 3.2.1 [matcher]
- riverpod 2.4.9 [meta stack_trace state_notifier]
- rxdart 0.27.7
- shared_preferences_android 2.2.1 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.3.5 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.3.2 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.3.2 [flutter plugin_platform_interface]
- shared_preferences_web 2.2.2 [flutter flutter_web_plugins shared_preferences_platform_interface web]
- shared_preferences_windows 2.3.2 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- sprintf 7.0.0
- sqflite 2.3.0 [flutter sqflite_common path]
- sqflite_common 2.5.0+2 [synchronized path meta]
- state_notifier 1.0.0 [meta]
- stream_channel 2.1.2 [async]
- string_scanner 1.2.0 [source_span]
- stripe_android 9.6.0+2 [flutter]
- stripe_ios 9.6.0 [flutter]
- stripe_js 3.3.0 [freezed_annotation js json_annotation meta]
- stripe_platform_interface 9.6.0 [flutter freezed_annotation json_annotation meta plugin_platform_interface]
- synchronized 3.1.0+1
- term_glyph 1.2.1
- test_api 0.6.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- typed_data 1.3.2 [collection]
- universal_html 2.2.4 [async csslib charcode collection html meta source_span typed_data universal_io]
- universal_io 2.2.2 [collection meta typed_data]
- universal_platform 1.0.0+1
- url_launcher_android 6.2.2 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.2.4 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.1.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.1.0 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.3.1 [flutter plugin_platform_interface]
- url_launcher_web 2.2.3 [flutter flutter_web_plugins url_launcher_platform_interface web]
- url_launcher_windows 3.1.1 [flutter url_launcher_platform_interface]
- uuid 4.3.3 [crypto sprintf meta fixnum]
- vector_graphics 1.1.9+2 [flutter vector_graphics_codec]
- vector_graphics_codec 1.1.9+2
- vector_graphics_compiler 1.1.9+2 [args meta path_parsing xml vector_graphics_codec path]
- vector_math 2.1.4
- video_player_android 2.4.11 [flutter video_player_platform_interface]
- video_player_avfoundation 2.5.6 [flutter video_player_platform_interface]
- video_player_platform_interface 6.2.2 [flutter plugin_platform_interface]
- wakelock_plus 1.1.4 [flutter flutter_web_plugins meta wakelock_plus_platform_interface win32 dbus package_info_plus js]
- wakelock_plus_platform_interface 1.1.0 [flutter plugin_platform_interface meta]
- web 0.3.0
- webview_flutter_android 3.13.2 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 2.9.1 [flutter meta plugin_platform_interface]
- webview_flutter_wkwebview 3.10.2 [flutter path webview_flutter_platform_interface]
- win32 5.2.0 [ffi]
- xdg_directories 1.0.4 [meta path]
- xml 6.5.0 [collection meta petitparser]

```

</details>

---

**Comment by darshankawar** [comment]
> after upgrading the flutter SDK to (3.16.9)

@rahmanrezaee  What version you were on before upgrading ?



> and shows an unclear error

Can you provide complete error log ?

Are you getting this behavior locally in debug mode with the code sample provided ?



> ```dart
> ReCaptchaEnterpriseProvider(AppConstants.enterpriseRecapKey),
> ```

Is it possible for you to use another Recaptcha provider (ex: ReCaptchaV3Provider) and see if using it, you still get same error or not ?

**Comment by rahmanrezaee** [comment]
@darshankawar I think I was using 3.16.6 

THis is my log after running
```
[   +8 ms] Debug service listening on ws://127.0.0.1:51074/IXHVRfO9GgU=/ws
[   +4 ms]   To hot restart changes while running, press "r" or "R".
[   +1 ms] For a more detailed help message, press "h". To quit, press "q".
[   +2 ms] A Dart VM Service on Chrome is available at: http://127.0.0.1:51074/IXHVRfO9GgU=
[  +67 ms] ExpressionEvaluator: Evaluating "() { return true; }" at packages/flutter/src/widgets/unique_widget.dart
[ +203 ms] ExpressionEvaluator: Evaluating JS: "function () {
                      try {
                        return   (function() {
                            const dart_sdk = require('dart_sdk');
                            const core = dart_sdk.core;
                            const dart = dart_sdk.dart;
                            var T = {
                              VoidTobool: () => (T.VoidTobool = dart.constFn(dart.fnType(core.bool, [])))()
                            };
                            return dart.fn(() => true, T.VoidTobool());
                          }(

                          ))();
                      } catch (error) {
                        return error.name + ": " + error.message;
                      }
                    }" with scope: {}
[  +12 ms] ExpressionEvaluator: Evaluated "() { return true; }" to "{type: boolean, value: true}"
[  +26 ms] The webOnlyAssetManager getter is deprecated and will be removed in a future release. Please use `assetManager` from `dart:ui_web` instead.
[   +3 ms] The webOnlyAssetManager getter is deprecated and will be removed in a future release. Please use `assetManager` from `dart:ui_web` instead.
[ +274 ms] The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:51074/IXHVRfO9GgU=
[+7258 ms] [2024-02-01T04:36:21.895Z]  @firebase/app-check:
[   +2 ms] FirebaseError: AppCheck: ReCAPTCHA error. (appCheck/recaptcha-error).

```

**Comment by rahmanrezaee** [comment]
@darshankawar 
I forgot to add this line while i removed the localhost domain from the recapture panel 😂
![image](https://github.com/firebase/flutterfire/assets/37028113/26d3a3c2-86c8-4384-90f8-8b1b17382153)
